helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -945,23 +947,23 @@ public void setInstanceConfig(String clusterName, String instanceName,
    * replaced with the value of the same field in given config if it presents. If there is new field
    * in given config but not in current config, the field will be added into the current config..
    * The list fields and map fields will be replaced as a single entry.
-   * The current Cluster config will be replaced with the given clusterConfig. WARNING: This is not
+   * The current instanceConfig will be replaced with the given instanceConfig. WARNING: This is not
    * thread-safe or concurrent updates safe.
    * *
    *
    * @param clusterName
    * @param instanceName
-   * @param instanceConfig
+   * @param newInstanceConfig
    *
    * @return
    */
   public void updateInstanceConfig(String clusterName, String instanceName,
-      InstanceConfig instanceConfig) {
-    updateInstanceConfig(clusterName, instanceName, instanceConfig, false);
+      InstanceConfig newInstanceConfig) {
```

Comment:
It should be okey to leave it "instanceConfig" instead of "newInstanceConfig", otherwise, you need to rename this parameter in all other "set" or "update" related methods to keep them consistent, which I do not recommend though.

Reply:
Updated,